apps/examples/src/shared/TipTap.tsx

```diff
@@ -0,0 +1,155 @@
+	fontSize: number
+	lineHeight: number
+	align: TLDefaultHorizontalAlignStyle
+	verticalAlign: TLDefaultVerticalAlignStyle
+	wrap?: boolean
+	bounds?: Box
+}) => {
+	const theme = useDefaultColorTheme()
+
+	const tldrawEditor = useTldrawEditor()
+	const isEditing = useValue('isEditing', () => tldrawEditor.getEditingShapeId() === id, [
+		tldrawEditor,
+		id,
+	])
+	const tiptapEditor = useTipTapEditor({
+		extensions,
+		content: content ? JSON.parse(content) : '',
+		autofocus: true,
+		onUpdate: ({ editor }) => {
+			const json = editor.getJSON()
+			tldrawEditor.updateShapes<TLUnknownShape & { props: { text: string } }>([
+				{ id, type, props: { text: JSON.stringify(json) } },
+			])
+		},
+		onBlur: () => {
+			tldrawEditor.complete()
+		},
+	})
+
+	useEffect(() => {
+		if (content && JSON.stringify(tiptapEditor?.getJSON()) !== content) {
+			tiptapEditor?.commands.setContent(content)
+		}
+	}, [tiptapEditor, content])
+
+	useEffect(() => {
+		if (isEditing && !tiptapEditor?.isFocused) {
+			tiptapEditor?.commands.focus('end')
+		}
+	}, [tiptapEditor, isEditing])
+
+	const handleInputPointerDown = useCallback(
+		(e: React.PointerEvent) => {
+			tldrawEditor.dispatch({
+				...getPointerInfo(e),
+				type: 'pointer',
+				name: 'pointer_down',
+				target: 'shape',
+				shape: tldrawEditor.getShape(id)!,
+			})
+
+			stopEventPropagation(e) // we need to prevent blurring the input
+		},
+		[tldrawEditor, id]
+	)
+
+	return (
+		<div
+			key={`tl-tiptap-${isEditing}`}
+			className="tl-text-label tl-text-wrapper"
+			data-font={font}
+			data-align={align}
+			data-textwrap={!!wrap}
+			style={{
+				pointerEvents: isEditing ? 'all' : 'none',
+				userSelect: isEditing ? 'all' : 'none',
+				justifyContent: align === 'middle' ? 'center' : align,
+				alignItems: verticalAlign === 'middle' ? 'center' : verticalAlign,
+				...(bounds
+					? {
+							top: bounds.minY,
+							left: bounds.minX,
+							width: bounds.width,
+							height: bounds.height,
+							position: 'absolute',
+						}
+					: {}),
+			}}
+		>
+			<div
+				className="tl-text-label__inner"
+				style={{
+					pointerEvents: isEditing ? 'all' : 'none',
+					userSelect: isEditing ? 'all' : 'none',
+					fontSize,
+					lineHeight: fontSize * lineHeight + 'px',
+					minHeight: lineHeight + 32,
+					minWidth: 0,
+					color: theme[labelColor as TLDefaultColorStyle].solid,
+				}}
+				onPointerDown={handleInputPointerDown}
+			>
+				<EditorContent editor={tiptapEditor} />
+			</div>
+		</div>
+	)
+}
+
+// This strictly renders just so we can get an accurate size of the div being rendered.
+export function TipTapMeasure({ content }: { content: string }) {
+	const tiptapEditor = useTipTapEditor({
+		extensions: [StarterKit],
+		content: content ? JSON.parse(content) : '',
+	})
+
+	useEffect(() => {
+		if (content && JSON.stringify(tiptapEditor?.getJSON()) !== content) {
+			tiptapEditor?.commands.setContent(JSON.parse(content))
+		}
+	}, [tiptapEditor, content])
```

Comment:
```suggestion
	const tiptapEditor = useTipTapEditor({
		extensions: [StarterKit],
		content: content ? JSON.parse(content) : '',
	}, [content])
```

useEditor takes a second parameter, a dep array that will re-render the editor from scratch when it's values change